cmake/modules/HCT.cmake

```diff
@@ -49,6 +58,17 @@ function(add_hlsl_hctgen mode)
                        ${hctdb}
                        ${hctdb_helper})
 
+  get_filename_component(output_extension ${full_output} LAST_EXT)
+
+  if (output_extension MATCHES "\.h|\.cpp|\.inl")
+    set(format_cmd COMMAND ${CLANG_FORMAT_EXE} -i ${temp_output})
+    if (NOT CLANG_FORMAT_EXE)
+      add_custom_target(${mode} DEPENDS ${output})
+      add_dependencies(HCTGen ${mode})
```

Comment:
What does this do to mitigate the absence of clang-format?

Reply:
It creates the generation targets as empty targets so that other places that depend on the targets don’t error out. Earlier on it logs a earring for missing clang-format, and will error if you don’t have clang format and explicitly set the option to re-generate the generated sources.